File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();

        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0)+1);
        }

        List<int[]> v = new ArrayList<>();

        for(Map.Entry<Integer, Integer> entry : map.entrySet()){
            v.add(new int[]{entry.getKey(), entry.getValue()});
        }

        Collections.sort(v, (a, b) -> b[1]-a[1]);

        int[] ans = new int[k];

        for(int i=0; i<k; i++){
            ans[i] = v.get(i)[0];
        }

        return ans;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int n = nums.size();
        vector<vector<int>> ans;

        sort(nums.begin(), nums.end());

        for(int i=0; i<n; i++){
            if (i > 0 && nums[i] == nums[i - 1]) continue;
            int low = i+1, high = n-1;

            int target = -nums[i];

            while(low < high){
                int sum = nums[low]+nums[high];
                if(sum==target){
                    ans.push_back({nums[i], nums[low], nums[high]});
                    
                    while(low < high && nums[low]==nums[low+1]) low++;
                    while(low < high && nums[high]==nums[high-1]) high--;

                    low++;
                    high--;
                }
                else if(sum < target) low++;
                else high--;
            }
        }

        return ans;
    }
};
